File: oop/point-v2.java
import java.lang.Math;
import java.util.Scanner;

class Point {
  public double Distance (Point p) {
    double dx = x - p.x, dy = y - p.y;
    return Math.sqrt(dx * dx + dy * dy);
  }
  public Point(double x, double y) {
    this.x = x;
    this.y = y;
  }
  public void Scale(double k) {
    x *= k;
    y *= k;
  }
  private double x, y;
}

class Main {
  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);
    Point p1 = new Point(sc.nextDouble(), sc.nextDouble()),
      p2 = new Point(sc.nextDouble(), sc.nextDouble());
    System.out.println(p1.Distance(p2));
  }
}

File: oop/point-v2.go
package main
import (
	"fmt"
	"math"
)

type point struct {
	x, y float64
}

func (p1 point) distance(p2 point) float64 {
	dx := p1.x - p2.x
	dy := p1.y - p2.y
	return math.Sqrt(dx * dx + dy * dy)
}

func (p point) scale(k float64) {
	p.x *= k
	p.y *= k
}

func main() {
	var p1, p2 point
	fmt.Scanf("%f%f%f%f", &p1.x, &p1.y,
		&p2.x, &p2.y)
	fmt.Printf("%f\n", p1.distance(p2))
}